sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -406,18 +445,24 @@ def size_in_bytes(self):
         return self._size
 
     def add(self, event_data):
-        # type: (EventData) -> None
+        # type: (Union[EventData, AmqpAnnotatedMessage]) -> None
         """Try to add an EventData to the batch.
 
         The total size of an added event is the sum of its body, properties, etc.
         If this added size results in the batch exceeding the maximum batch size, a `ValueError` will
         be raised.
 
         :param event_data: The EventData to add to the batch.
-        :type event_data: ~azure.eventhub.EventData
+        :type event_data: Union[~azure.eventhub.EventData, ~azure.eventhub.amqp.AmqpAnnotatedMessage]
         :rtype: None
         :raise: :class:`ValueError`, when exceeding the size limit.
         """
+        # pylint:disable=protected-access
+        if isinstance(event_data, AmqpAnnotatedMessage):
+            event_data = EventData._from_message(event_data._to_outgoing_amqp_message())    # pylint: disable=protected-access
```

Comment:
is it possible to add the `_to_outgoing_message` on `AmqpAnnotatedMessage` like what we have done in Service Bus?

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py#L296-L298

Reply:
So I didn't add it because in the SB AmqpAnnotatedMessage `_to_outgoing_message`, you send in the `message` arg into `ServiceBusMessage`. However, `EventData` does not take keyword args, so we cannot pass in `message`. If we wanted to do this, we would need to allow for an optional `message` property on `EventData` and change the API. Thoughts?

Reply:
I think we ultimately would need the kwargs in the constructor.

I'm checking the [design guideline](https://azure.github.io/azure-sdk/python_design.html#python-models-input-constructor)
> DO craft a constructor for models that are intended to be instantiated by a user (i.e. non-result types) with minimal required information and optional information as keyword-only arguments.

We're missing it in EventData (e.g. `properties` could be keyword argument in the constructor), but that's a different topic.

I'm good with leaving as it is now as it's an implementation detail, I feel adding kwargs to the constructor while users could not pass anything and there being no documentation is not friendly.

@annatisch , what about your opinions?